# Parte 1: Gerar a matriz de atributos

In [1]:
# Conectando ao google Drive
import numpy as np
import matplotlib.pyplot as plt
import scipy.io as sp
from scipy.fft import fft
from google.colab import drive
import pandas as pd
from sklearn.model_selection import train_test_split

#Endereço da base de dados no Google Drive
path_database = '/content/gdrive/MyDrive/datasets_sinais_cerebrais/'

#Autoriza a leitura do Google Drive
drive.mount('/content/gdrive')

mat = sp.loadmat(path_database+'S20.mat')             
data = np.array(mat['data'])
print(np.shape(data))

Mounted at /content/gdrive
(64, 1500, 40, 6)


In [2]:
# implementado as funçoes

def Vcar(Matriz_freq):
  media = np.mean(Matriz_freq, axis=0)
  Vcar = Matriz_freq - media
  return Vcar

def Matriz_caracteristica(Matriz_freq, array_frequencias,frequencia_observada,coleta_atual,qtd_janelas,canal): #array_frequencias[8,10,12,15]
  n_freq = len(array_frequencias)
  H_FFT = np.empty((qtd_janelas,n_freq))
  
  intervalo = int(1250/qtd_janelas)
  pontos = 0
  for janela in range(qtd_janelas):
    mag_fft = np.abs(fft(Matriz_freq[canal,pontos:pontos+intervalo]))
    pontos = pontos + intervalo
    H_FFT[janela] = mag_fft[array_frequencias]
      
    #plotar_grafico((Matriz_freq[61,pontos:pontos+250]),mag_fft,janela,frequencia_observada,coleta_atual)
     
  return H_FFT

def plotar_grafico(array_no_tempo,mag_fft,janela,frequencia_observada,coleta):
  ff = np.arange(0, 50, 0.2) # uma trial, que coresponde a 1 segundo, que corresponde a 250 pontos - 50/0.2 = 250
  print("---------------- frequencia observada: ",frequencia_observada,"coleta: ",coleta, "Janela: ",janela,"----------------")
  plt.subplot(2,1,1)
  plt.plot(array_no_tempo)
  plt.subplot(2,1,2)
  plt.plot(ff, mag_fft)
  plt.xlim(5, 15)
  plt.ylim(0, 2200)
  plt.grid()
  plt.show()

In [3]:
#gerar matriz final de atributos:

coleta = [0,1,2,3,4,5]  # de 0 a 5
freq = [0,2,4,7] # 8hz, 10hz, 12hz, 15hz
frequencias = [8,10,12,15]
canais = [47,53,54,55,56,57,60,61,62]
qtd_janelas = 5 #5 janelas -> 1segundo cada janela, 1 janela -> 5 segundos, 2 janelas -> 625pts q equivale a 2,5 segundos cada janela

for canal in canais:
  for f in freq:
    for c in coleta:
      Matriz_freq = data[:, 125:1375,f,c] #matriz gerada quando o individuo esta olhando para uma determinada frequencia
      MatrizCar = Vcar(Matriz_freq) 
      aux = Matriz_caracteristica(MatrizCar,frequencias,f,c,qtd_janelas,canal) 
      if (c ==0):
        hfft = aux
      else:
        hfft = np.vstack((hfft,aux))
    if f==0:
      hfft_2 = hfft
    else:
      hfft_2 = np.vstack((hfft_2,hfft))
  if(canal==47):
    hfft_final = hfft_2
  else:
    hfft_final = np.hstack((hfft_final,hfft_2)) 



In [4]:
print("dimensão da matriz final de atributos: ",hfft_final.shape)
print('amostra 0:')
print(hfft_final[0])

dimensão da matriz final de atributos:  (120, 36)
amostra 0:
[ 35.388066   215.26140318  85.25346112  66.03489953 401.22658118
 130.49588873 222.42061976  66.24811954 394.74002855 141.70510426
 227.31337972  68.91392904 116.53228547 233.03921678 179.85146409
  94.20172037 174.76171647  70.20318312 257.01324999 165.10241103
 201.7454916   25.13440265 224.94771343 132.12129699 467.01155916
   2.58963462 255.51055597  67.93949132 439.70997846  76.8936721
 230.61165372  97.92357418 353.25194361  51.62520197 313.01585596
 111.18088305]


# Parte 2: Classifcação

In [7]:
from sklearn.preprocessing import StandardScaler
num_linhas = len(hfft_final)
#H = np.hstack((hfft_final, np.ones((num_linhas,1))))

#one hot encoder
l1 = np.hstack((np.ones((1,30)), np.zeros((1,90))))
l2 = np.hstack((np.zeros((1,30)), np.ones((1,30)), np.zeros((1,60))))
l3 = np.hstack((np.zeros((1,60)), np.ones((1,30)), np.zeros((1,30))))
l4 = np.hstack((np.zeros((1,90)), np.ones((1,30))))

l_aux = np.vstack((l1,l2))
l_aux2 = np.vstack((l_aux,l3))
label = np.vstack((l_aux2,l4)).T

 
scaler = StandardScaler()
scaler.fit(hfft_final)
H = scaler.transform(hfft_final)
A = np.hstack((H, np.ones((num_linhas,1))))
A = A*0.01
def holdOut():
  array_acc = []
  for i in range(20):
    #particao dos dados
    X_train, X_val, y_train, y_val = train_test_split(A, label, test_size=.20, random_state=None)


    #TREINANDO O CLASSIFICADOR
    numClasses = len(frequencias)
    w = np.zeros((37, numClasses), dtype=int)

    #Calcula o vetor de pesos (hiperplano) para cada classe
    for c in range(numClasses):
        w[:,c] = np.linalg.pinv(X_train, rcond=1e-15)@y_train[:,c]

    Y = np.zeros((X_val.shape[0],numClasses))
    #VALIDANDO O CLASSIFICADOR
    for c in range(numClasses):
      Y[:,c] = np.matmul(X_val, w[:,c])
      
    maxIndexY = np.argmax(Y, axis=1) # estima do classificador
    maxIndexY_val = np.argmax(y_val, axis=1) # resposta correta

    numCorrect=0
    numWrong = 0
    for index in range(maxIndexY.shape[0]):
      if(maxIndexY[index] == maxIndexY_val[index]):
        numCorrect = numCorrect + 1
      else:
        numWrong = numWrong + 1
    acc = numCorrect/(numCorrect+numWrong)
    array_acc.append(acc)
  array_acc = np.array(array_acc)
  return array_acc.mean(), array_acc.std()

media, std = holdOut()
media, std

(0.825, 0.070464727188699)

#Parte 3: Implementação da GAN

In [ ]:
import tensorflow as tf
import numpy as np
from tensorflow.keras import layers
import time

In [ ]:
class Gan:
  def __init__(self, epochs, noise_dim, dataset, gen_dim, lr):
    self.epochs = epochs
    self.noise_dim = noise_dim
    self.dataset = dataset
    self.gen_dim = gen_dim
    self.lr = lr
    self.cross_entropy = tf.keras.losses.BinaryCrossentropy(from_logits=True)
    self.generator = self.create_generator()
    self.discriminator = self.create_discriminator()
    self.D_cycle = 10

  def create_generator(self):
    model = tf.keras.Sequential()
    model.add(layers.Dense(256, input_shape=(self.noise_dim,)))
    model.add(layers.Dense(128))
    model.add(layers.Dense(64))
    model.add(layers.Dense(self.gen_dim, activation="relu"))
    
    return model

  def create_discriminator(self):
    model = tf.keras.Sequential()
    model.add(layers.Dense(128, input_shape=(self.gen_dim,)))
    model.add(layers.Dense(64))
    model.add(layers.Dense(36))
    model.add(layers.Dense(1, activation='softmax'))

    return model
    
  def D_loss(self, real_output, fake_output):
    real_loss = self.cross_entropy(tf.ones_like(real_output), real_output)
    fake_loss = self.cross_entropy(tf.zeros_like(fake_output), fake_output)
    total_loss = real_loss + fake_loss
    return total_loss
  
  def G_loss(self, fake_output):
    return self.cross_entropy(tf.ones_like(fake_output), fake_output)
  
  @tf.function
  def trainD(self, batch_samples):
    noise = tf.random.normal([batch_samples.shape[0], self.noise_dim])
    with tf.GradientTape() as disc_tape:
      generated_samples = self.generator(noise, training=True)
      real_output = self.discriminator(batch_samples, training=True)
      fake_output = self.discriminator(generated_samples, training=True)
      disc_loss = self.D_loss(real_output, fake_output)
    gradients_of_discriminator = disc_tape.gradient(disc_loss, self.discriminator.trainable_variables)
    self.discriminator_optimizer.apply_gradients(zip(gradients_of_discriminator, self.discriminator.trainable_variables))

    return disc_loss

  @tf.function
  def trainG(self, batch_samples):
    noise = tf.random.normal([batch_samples.shape[0], self.noise_dim])
    with tf.GradientTape() as gen_tape:
      generated_samples = self.generator(noise, training=True)
      fake_output = self.discriminator(generated_samples, training=True)
      gen_loss = self.G_loss(fake_output)
    gradients_of_generator = gen_tape.gradient(gen_loss, self.generator.trainable_variables)
    self.generator_optimizer.apply_gradients(zip(gradients_of_generator, self.generator.trainable_variables))

    return gen_loss

  def fit(self):
    lossG = []
    lossD = []
    self.generator_optimizer = tf.keras.optimizers.Adam(self.lr)
    self.discriminator_optimizer = tf.keras.optimizers.Adam(self.lr)
    for epoch in range(self.epochs):
      start = time.time()

      for batch_samples in self.dataset:
        Gloss = self.trainG(batch_samples)
        for _ in range(self.D_cycle):
          Dloss = self.trainD(batch_samples)
        
      lossG.append(Gloss.numpy())
      lossD.append(Dloss.numpy())
        
      print('Time for epoch {} is {} sec'.format(epoch + 1, time.time()-start))

In [ ]:
def avaliar_0_dados_reais(X_val,y_val,X_train,y_train):
  #normalização
  scaler1 = StandardScaler()
  scaler1.fit(X_train)
  X_train = scaler1.transform(X_train)
  X_val = scaler1.transform(X_val)
  

  X_train = np.hstack((X_train,np.ones((X_train.shape[0],1))))
  X_train = X_train*0.01
  X_val = np.hstack((X_val,np.ones((X_val.shape[0],1))))
  X_val = X_val*0.01

  #TREINANDO O CLASSIFICADOR
  numClasses = len(frequencias)
  w = np.zeros((37, numClasses), dtype=int)

  #Calcula o vetor de pesos (hiperplano) para cada classe
  for c in range(numClasses):
      w[:,c] = np.linalg.pinv(X_train, rcond=1e-15)@y_train[:,c]

  Y = np.zeros((X_val.shape[0],numClasses))
  #VALIDANDO O CLASSIFICADOR
  for c in range(numClasses):
    Y[:,c] = np.matmul(X_val, w[:,c])
    
  maxIndexY = np.argmax(Y, axis=1) # estima do classificador
  maxIndexY_val = np.argmax(y_val, axis=1) # resposta correta

  numCorrect=0
  numWrong = 0
  for index in range(maxIndexY.shape[0]):
    if(maxIndexY[index] == maxIndexY_val[index]):
      numCorrect = numCorrect + 1
    else:
      numWrong = numWrong + 1

  acuracia = numCorrect/(numCorrect+numWrong) 

  #matriz de confusão:
  y_true = maxIndexY_val
  y_pred = maxIndexY
  matriz_de_confusao = multilabel_confusion_matrix(y_true, y_pred)

  return acuracia, matriz_de_confusao


In [ ]:
# funçoes auxiliares para a validação
def generate_synthetic_samples(data, qtd_samples,rotulo):
  dataset = tf.data.Dataset.from_tensor_slices(data).shuffle(data.shape[0]).batch(4)
  noise_dim = 8

  gan = Gan(epochs=150, noise_dim=noise_dim, dataset=dataset,gen_dim=36, lr=1e-4)
  gan.fit();
  noise = tf.random.normal(shape=(qtd_samples, noise_dim))
  synthetic_samples = gan.generator(noise)

  y = np.zeros((qtd_samples,4))
  y[:,rotulo] = 1.0

  return synthetic_samples.numpy(), y, gan


from sklearn.metrics import multilabel_confusion_matrix
from sklearn.preprocessing import StandardScaler
def avaliar_desempenho(X_train, y_train, X_val, y_val, X_synthetic_samples, y_synthetic_samples):

   #normalização
  scaler1 = StandardScaler()
  scaler1.fit(X_train)
  X_train = scaler1.transform(X_train)
  X_val = scaler1.transform(X_val)
  
  if (X_synthetic_samples.shape[0]!=0):
    scaler2 = StandardScaler()
    scaler2.fit(X_synthetic_samples)
    X_synthetic_samples = scaler2.transform(X_synthetic_samples)  
    X_train = np.vstack((X_train,X_synthetic_samples))
    y_train = np.vstack((y_train,y_synthetic_samples))

  X_train = np.hstack((X_train,np.ones((X_train.shape[0],1))))
  X_train = X_train*0.01
  X_val = np.hstack((X_val,np.ones((X_val.shape[0],1))))
  X_val = X_val*0.01

  #TREINANDO O CLASSIFICADOR
  numClasses = len(frequencias)
  w = np.zeros((37, numClasses), dtype=int)

  #Calcula o vetor de pesos (hiperplano) para cada classe
  for c in range(numClasses):
      w[:,c] = np.linalg.pinv(X_train, rcond=1e-15)@y_train[:,c]

  Y = np.zeros((X_val.shape[0],numClasses))
  #VALIDANDO O CLASSIFICADOR
  for c in range(numClasses):
    Y[:,c] = np.matmul(X_val, w[:,c])
    
  maxIndexY = np.argmax(Y, axis=1) # estima do classificador
  maxIndexY_val = np.argmax(y_val, axis=1) # resposta correta

  numCorrect=0
  numWrong = 0
  for index in range(maxIndexY.shape[0]):
    if(maxIndexY[index] == maxIndexY_val[index]):
      numCorrect = numCorrect + 1
    else:
      numWrong = numWrong + 1

  acuracia = numCorrect/(numCorrect+numWrong) 

  #matriz de confusão:
  y_true = maxIndexY_val
  y_pred = maxIndexY
  matriz_de_confusao = multilabel_confusion_matrix(y_true, y_pred)

  return acuracia, matriz_de_confusao


# validação

In [ ]:
l1 = np.hstack((np.ones((1,30)), np.zeros((1,90))))
l2 = np.hstack((np.zeros((1,30)), np.ones((1,30)), np.zeros((1,60))))
l3 = np.hstack((np.zeros((1,60)), np.ones((1,30)), np.zeros((1,30))))
l4 = np.hstack((np.zeros((1,90)), np.ones((1,30))))
l_aux = np.vstack((l1,l2))
l_aux2 = np.vstack((l_aux,l3))
y_real = np.vstack((l_aux2,l4)).T


def validar(n_synthetic_samples, rs):
  acuracia = []
  for i in range(20):
    y = np.zeros((30,4))
    y[:,0] = 1
    X_train_8hz, X_val_8hz, y_train_8hz, y_val_8hz = train_test_split(hfft_final[0:30,:], y, test_size=.20, random_state=rs)
    y = np.zeros((30,4))
    y[:,1] = 1
    X_train_10hz, X_val_10hz, y_train_10hz, y_val_10hz = train_test_split(hfft_final[30:60,:], y, test_size=.20, random_state=rs)
    y = np.zeros((30,4))
    y[:,2] = 1
    X_train_12hz, X_val_12hz, y_train_12hz, y_val_12hz = train_test_split(hfft_final[60:90,:], y, test_size=.20, random_state=rs)
    y = np.zeros((30,4))
    y[:,3] = 1
    X_train_15hz, X_val_15hz, y_train_15hz, y_val_15hz = train_test_split(hfft_final[90:120,:], y, test_size=.20, random_state=rs)

    indices = np.random.randint(24, size=(12))
    X_mix_bci = X_train_8hz[indices]
    y_mix_bci = y_train_8hz[indices]

    indices = np.random.randint(24, size=(12))
    X_mix_bci = np.vstack((X_mix_bci,X_train_10hz[indices]))
    y_mix_bci = np.vstack((y_mix_bci,y_train_10hz[indices]))

    indices = np.random.randint(24, size=(12))
    X_mix_bci = np.vstack((X_mix_bci,X_train_12hz[indices]))
    y_mix_bci = np.vstack((y_mix_bci,y_train_12hz[indices]))

    indices = np.random.randint(24, size=(12))
    X_mix_bci = np.vstack((X_mix_bci,X_train_15hz[indices]))
    y_mix_bci = np.vstack((y_mix_bci,y_train_15hz[indices]))

    X_train_bci = np.vstack((X_train_8hz,X_train_10hz))
    X_train_bci = np.vstack((X_train_bci,X_train_12hz))
    X_train_bci = np.vstack((X_train_bci,X_train_15hz))

    y_train_bci = np.vstack((y_train_8hz,y_train_10hz))
    y_train_bci = np.vstack((y_train_bci,y_train_12hz))
    y_train_bci = np.vstack((y_train_bci,y_train_15hz))

    X_val_bci = np.vstack((X_val_8hz,X_val_10hz))
    X_val_bci = np.vstack((X_val_bci,X_val_12hz))
    X_val_bci = np.vstack((X_val_bci,X_val_15hz))

    y_val_bci = np.vstack((y_val_8hz,y_val_10hz))
    y_val_bci = np.vstack((y_val_bci,y_val_12hz))
    y_val_bci = np.vstack((y_val_bci,y_val_15hz))


    synthetic_samples_8hz, y_8hz, gan_8hz = generate_synthetic_samples(X_train_8hz, n_synthetic_samples,0)
    synthetic_samples_10hz, y_10hz, gan_10hz = generate_synthetic_samples(X_train_10hz, n_synthetic_samples,1)
    synthetic_samples_12hz, y_12hz, gan_12hz = generate_synthetic_samples(X_train_12hz, n_synthetic_samples,2)
    synthetic_samples_15hz, y_15hz, gan_15hz = generate_synthetic_samples(X_train_15hz, n_synthetic_samples,3)

    X_synthetic_samples = np.vstack((synthetic_samples_8hz,synthetic_samples_10hz))
    X_synthetic_samples = np.vstack((X_synthetic_samples,synthetic_samples_12hz))
    X_synthetic_samples = np.vstack((X_synthetic_samples,synthetic_samples_15hz))

    y_synthetic_samples = np.vstack((y_8hz,y_10hz))
    y_synthetic_samples = np.vstack((y_synthetic_samples,y_12hz))
    y_synthetic_samples = np.vstack((y_synthetic_samples,y_15hz))

    

    X_mix_bci = np.vstack((X_mix_bci,X_synthetic_samples))
    y_mix_bci = np.vstack((y_mix_bci,y_synthetic_samples))


    ac,mc = avaliar_desempenho(X_train_bci, y_train_bci, X_val_bci, y_val_bci, X_synthetic_samples, y_synthetic_samples)
    #ac,mc = avaliar_0_dados_reais(X_val_bci,y_val_bci,X_synthetic_samples,y_synthetic_samples)
    acuracia.append(ac)
  acuracia = np.array(acuracia)
  return acuracia.mean(),acuracia.std(),acuracia

In [ ]:
testar = [4,6]
array_media = []
array_std = []
array_acc = []
inicio = time.time()
for x in testar:
  media,std,acc = validar(x, None)
  array_media.append(media)
  array_std.append(std)
  array_acc.append(acc)
fim = time.time()
tempo_gasto = fim - inicio


/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 3.941063642501831 sec
Time for epoch 2 is 0.14205360412597656 sec
Time for epoch 3 is 0.1429135799407959 sec
Time for epoch 4 is 0.10698986053466797 sec
Time for epoch 5 is 0.1422569751739502 sec
Time for epoch 6 is 0.1430051326751709 sec
Time for epoch 7 is 0.1430799961090088 sec
Time for epoch 8 is 0.14325261116027832 sec
Time for epoch 9 is 0.1268167495727539 sec
Time for epoch 10 is 0.1431126594543457 sec
Time for epoch 11 is 0.1427624225616455 sec
Time for epoch 12 is 0.14317727088928223 sec
Time for epoch 13 is 0.14308738708496094 sec
Time for epoch 14 is 0.14180397987365723 sec
Time for epoch 15 is 0.1437549591064453 sec
Time for epoch 16 is 0.1093602180480957 sec
Time for epoch 17 is 0.13640832901000977 sec
Time for epoch 18 is 0.1422581672668457 sec
Time for epoch 19 is 0.14297938346862793 sec
Time for epoch 20 is 0.14147257804870605 sec
Time for epoch 21 is 0.14406633377075195 sec
Time for epoch 22 is 0.10395932197570801 sec
Time for epoch 23 is 0.14331865

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 2.689455509185791 sec
Time for epoch 2 is 0.14314794540405273 sec
Time for epoch 3 is 0.09635114669799805 sec
Time for epoch 4 is 0.08939814567565918 sec
Time for epoch 5 is 0.14179658889770508 sec
Time for epoch 6 is 0.1424260139465332 sec
Time for epoch 7 is 0.14339995384216309 sec
Time for epoch 8 is 0.14305520057678223 sec
Time for epoch 9 is 0.1418321132659912 sec
Time for epoch 10 is 0.14300799369812012 sec
Time for epoch 11 is 0.10138273239135742 sec
Time for epoch 12 is 0.14096975326538086 sec
Time for epoch 13 is 0.10006427764892578 sec
Time for epoch 14 is 0.1421976089477539 sec
Time for epoch 15 is 0.14417409896850586 sec
Time for epoch 16 is 0.14173293113708496 sec
Time for epoch 17 is 0.1426537036895752 sec
Time for epoch 18 is 0.1428818702697754 sec
Time for epoch 19 is 0.14232349395751953 sec
Time for epoch 20 is 0.09650492668151855 sec
Time for epoch 21 is 0.1425313949584961 sec
Time for epoch 22 is 0.14183616638183594 sec
Time for epoch 23 is 0.1424

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 2.7797107696533203 sec
Time for epoch 2 is 0.14375734329223633 sec
Time for epoch 3 is 0.09371733665466309 sec
Time for epoch 4 is 0.14273881912231445 sec
Time for epoch 5 is 0.14428234100341797 sec
Time for epoch 6 is 0.14109086990356445 sec
Time for epoch 7 is 0.10013389587402344 sec
Time for epoch 8 is 0.10345745086669922 sec
Time for epoch 9 is 0.1415109634399414 sec
Time for epoch 10 is 0.09984064102172852 sec
Time for epoch 11 is 0.09647607803344727 sec
Time for epoch 12 is 0.1102898120880127 sec
Time for epoch 13 is 0.09377789497375488 sec
Time for epoch 14 is 0.14199495315551758 sec
Time for epoch 15 is 0.10369491577148438 sec
Time for epoch 16 is 0.14306187629699707 sec
Time for epoch 17 is 0.0926675796508789 sec
Time for epoch 18 is 0.1428234577178955 sec
Time for epoch 19 is 0.14272379875183105 sec
Time for epoch 20 is 0.14180946350097656 sec
Time for epoch 21 is 0.14300823211669922 sec
Time for epoch 22 is 0.09630489349365234 sec
Time for epoch 23 is 0.0

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 5.111103296279907 sec
Time for epoch 2 is 0.14210176467895508 sec
Time for epoch 3 is 0.10233259201049805 sec
Time for epoch 4 is 0.14228463172912598 sec
Time for epoch 5 is 0.09944725036621094 sec
Time for epoch 6 is 0.143510103225708 sec
Time for epoch 7 is 0.14426088333129883 sec
Time for epoch 8 is 0.14257335662841797 sec
Time for epoch 9 is 0.1425952911376953 sec
Time for epoch 10 is 0.14226508140563965 sec
Time for epoch 11 is 0.14554357528686523 sec
Time for epoch 12 is 0.14301347732543945 sec
Time for epoch 13 is 0.09769153594970703 sec
Time for epoch 14 is 0.1419227123260498 sec
Time for epoch 15 is 0.11294198036193848 sec
Time for epoch 16 is 0.1057591438293457 sec
Time for epoch 17 is 0.1001577377319336 sec
Time for epoch 18 is 0.10034894943237305 sec
Time for epoch 19 is 0.09277558326721191 sec
Time for epoch 20 is 0.09430551528930664 sec
Time for epoch 21 is 0.10058307647705078 sec
Time for epoch 22 is 0.14156389236450195 sec
Time for epoch 23 is 0.1422

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 2.8735718727111816 sec
Time for epoch 2 is 0.1422100067138672 sec
Time for epoch 3 is 0.14210915565490723 sec
Time for epoch 4 is 0.1415388584136963 sec
Time for epoch 5 is 0.1412947177886963 sec
Time for epoch 6 is 0.09091901779174805 sec
Time for epoch 7 is 0.09774041175842285 sec
Time for epoch 8 is 0.14242911338806152 sec
Time for epoch 9 is 0.14290642738342285 sec
Time for epoch 10 is 0.14180254936218262 sec
Time for epoch 11 is 0.14181256294250488 sec
Time for epoch 12 is 0.09767818450927734 sec
Time for epoch 13 is 0.1415696144104004 sec
Time for epoch 14 is 0.09601736068725586 sec
Time for epoch 15 is 0.14110469818115234 sec
Time for epoch 16 is 0.09913516044616699 sec
Time for epoch 17 is 0.1416027545928955 sec
Time for epoch 18 is 0.30605506896972656 sec
Time for epoch 19 is 0.30300259590148926 sec
Time for epoch 20 is 0.3030085563659668 sec
Time for epoch 21 is 0.3039567470550537 sec
Time for epoch 22 is 0.3023815155029297 sec
Time for epoch 23 is 0.14821

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 5.108952760696411 sec
Time for epoch 2 is 0.14255809783935547 sec
Time for epoch 3 is 0.12181711196899414 sec
Time for epoch 4 is 0.1070404052734375 sec
Time for epoch 5 is 0.14251494407653809 sec
Time for epoch 6 is 0.0993199348449707 sec
Time for epoch 7 is 0.09757280349731445 sec
Time for epoch 8 is 0.14280056953430176 sec
Time for epoch 9 is 0.10973858833312988 sec
Time for epoch 10 is 0.14208006858825684 sec
Time for epoch 11 is 0.11463212966918945 sec
Time for epoch 12 is 0.1418473720550537 sec
Time for epoch 13 is 0.10939908027648926 sec
Time for epoch 14 is 0.1418898105621338 sec
Time for epoch 15 is 0.14223504066467285 sec
Time for epoch 16 is 0.14119243621826172 sec
Time for epoch 17 is 0.14136886596679688 sec
Time for epoch 18 is 0.14293265342712402 sec
Time for epoch 19 is 0.10125565528869629 sec
Time for epoch 20 is 0.1415414810180664 sec
Time for epoch 21 is 0.3048563003540039 sec
Time for epoch 22 is 0.3094451427459717 sec
Time for epoch 23 is 0.30330

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 5.108208894729614 sec
Time for epoch 2 is 0.14205503463745117 sec
Time for epoch 3 is 0.14243817329406738 sec
Time for epoch 4 is 0.09217095375061035 sec
Time for epoch 5 is 0.14213871955871582 sec
Time for epoch 6 is 0.1423184871673584 sec
Time for epoch 7 is 0.1042947769165039 sec
Time for epoch 8 is 0.10168170928955078 sec
Time for epoch 9 is 0.14227867126464844 sec
Time for epoch 10 is 0.17096829414367676 sec
Time for epoch 11 is 0.3097419738769531 sec
Time for epoch 12 is 0.3021562099456787 sec
Time for epoch 13 is 0.14286398887634277 sec
Time for epoch 14 is 0.30293798446655273 sec
Time for epoch 15 is 0.14195489883422852 sec
Time for epoch 16 is 0.15619349479675293 sec
Time for epoch 17 is 0.3035414218902588 sec
Time for epoch 18 is 0.30360841751098633 sec
Time for epoch 19 is 0.30399560928344727 sec
Time for epoch 20 is 0.1431434154510498 sec
Time for epoch 21 is 0.1445472240447998 sec
Time for epoch 22 is 0.13274002075195312 sec
Time for epoch 23 is 0.14679

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 2.84091854095459 sec
Time for epoch 2 is 0.14226126670837402 sec
Time for epoch 3 is 0.1426539421081543 sec
Time for epoch 4 is 0.15966343879699707 sec
Time for epoch 5 is 0.302081823348999 sec
Time for epoch 6 is 0.1423180103302002 sec
Time for epoch 7 is 0.30428576469421387 sec
Time for epoch 8 is 0.3048739433288574 sec
Time for epoch 9 is 0.1466813087463379 sec
Time for epoch 10 is 0.3018796443939209 sec
Time for epoch 11 is 0.13916277885437012 sec
Time for epoch 12 is 0.3049924373626709 sec
Time for epoch 13 is 0.30380988121032715 sec
Time for epoch 14 is 0.14220857620239258 sec
Time for epoch 15 is 0.30358123779296875 sec
Time for epoch 16 is 0.3035886287689209 sec
Time for epoch 17 is 0.30354857444763184 sec
Time for epoch 18 is 0.09451794624328613 sec
Time for epoch 19 is 0.14197182655334473 sec
Time for epoch 20 is 0.14262127876281738 sec
Time for epoch 21 is 0.14220881462097168 sec
Time for epoch 22 is 0.09629940986633301 sec
Time for epoch 23 is 0.14151167

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 2.8447282314300537 sec
Time for epoch 2 is 0.09678339958190918 sec
Time for epoch 3 is 0.09845614433288574 sec
Time for epoch 4 is 0.09737968444824219 sec
Time for epoch 5 is 0.10308670997619629 sec
Time for epoch 6 is 0.10615921020507812 sec
Time for epoch 7 is 0.09935545921325684 sec
Time for epoch 8 is 0.10891532897949219 sec
Time for epoch 9 is 0.09183502197265625 sec
Time for epoch 10 is 0.0941014289855957 sec
Time for epoch 11 is 0.14337873458862305 sec
Time for epoch 12 is 0.09532785415649414 sec
Time for epoch 13 is 0.14768648147583008 sec
Time for epoch 14 is 0.309246301651001 sec
Time for epoch 15 is 0.14246535301208496 sec
Time for epoch 16 is 0.30391621589660645 sec
Time for epoch 17 is 0.13596224784851074 sec
Time for epoch 18 is 0.30562520027160645 sec
Time for epoch 19 is 0.30336666107177734 sec
Time for epoch 20 is 0.3026614189147949 sec
Time for epoch 21 is 0.15140509605407715 sec
Time for epoch 22 is 0.30291271209716797 sec
Time for epoch 23 is 0.1

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 5.108961343765259 sec
Time for epoch 2 is 0.15285706520080566 sec
Time for epoch 3 is 0.3021245002746582 sec
Time for epoch 4 is 0.30379605293273926 sec
Time for epoch 5 is 0.30376505851745605 sec
Time for epoch 6 is 0.14372467994689941 sec
Time for epoch 7 is 0.14998936653137207 sec
Time for epoch 8 is 0.3023185729980469 sec
Time for epoch 9 is 0.08968091011047363 sec
Time for epoch 10 is 0.14146637916564941 sec
Time for epoch 11 is 0.09713268280029297 sec
Time for epoch 12 is 0.14236092567443848 sec
Time for epoch 13 is 0.10250186920166016 sec
Time for epoch 14 is 0.1423795223236084 sec
Time for epoch 15 is 0.14142274856567383 sec
Time for epoch 16 is 0.09845280647277832 sec
Time for epoch 17 is 0.10211491584777832 sec
Time for epoch 18 is 0.1417543888092041 sec
Time for epoch 19 is 0.1427443027496338 sec
Time for epoch 20 is 0.10175871849060059 sec
Time for epoch 21 is 0.14162206649780273 sec
Time for epoch 22 is 0.14265084266662598 sec
Time for epoch 23 is 0.142

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 5.109117746353149 sec
Time for epoch 2 is 0.3026437759399414 sec
Time for epoch 3 is 0.16048789024353027 sec
Time for epoch 4 is 0.1411914825439453 sec
Time for epoch 5 is 0.1427934169769287 sec
Time for epoch 6 is 0.14167118072509766 sec
Time for epoch 7 is 0.14228034019470215 sec
Time for epoch 8 is 0.10236668586730957 sec
Time for epoch 9 is 0.12211108207702637 sec
Time for epoch 10 is 0.10596704483032227 sec
Time for epoch 11 is 0.14329814910888672 sec
Time for epoch 12 is 0.14196300506591797 sec
Time for epoch 13 is 0.14150357246398926 sec
Time for epoch 14 is 0.14271235466003418 sec
Time for epoch 15 is 0.1412951946258545 sec
Time for epoch 16 is 0.14338970184326172 sec
Time for epoch 17 is 0.142106294631958 sec
Time for epoch 18 is 0.1506648063659668 sec
Time for epoch 19 is 0.09589505195617676 sec
Time for epoch 20 is 0.09711074829101562 sec
Time for epoch 21 is 0.14224600791931152 sec
Time for epoch 22 is 0.14239978790283203 sec
Time for epoch 23 is 0.14248

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 5.108935356140137 sec
Time for epoch 2 is 0.14287638664245605 sec
Time for epoch 3 is 0.11600399017333984 sec
Time for epoch 4 is 0.14187145233154297 sec
Time for epoch 5 is 0.10217785835266113 sec
Time for epoch 6 is 0.10644268989562988 sec
Time for epoch 7 is 0.09092354774475098 sec
Time for epoch 8 is 0.09831833839416504 sec
Time for epoch 9 is 0.14292216300964355 sec
Time for epoch 10 is 0.14278769493103027 sec
Time for epoch 11 is 0.14232110977172852 sec
Time for epoch 12 is 0.1452617645263672 sec
Time for epoch 13 is 0.09669804573059082 sec
Time for epoch 14 is 0.14090538024902344 sec
Time for epoch 15 is 0.1427929401397705 sec
Time for epoch 16 is 0.09827208518981934 sec
Time for epoch 17 is 0.14204955101013184 sec
Time for epoch 18 is 0.14149713516235352 sec
Time for epoch 19 is 0.14191842079162598 sec
Time for epoch 20 is 0.30281996726989746 sec
Time for epoch 21 is 0.1421825885772705 sec
Time for epoch 22 is 0.30155134201049805 sec
Time for epoch 23 is 0.3

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 5.108934640884399 sec
Time for epoch 2 is 0.14349675178527832 sec
Time for epoch 3 is 0.1426546573638916 sec
Time for epoch 4 is 0.09187531471252441 sec
Time for epoch 5 is 0.10108566284179688 sec
Time for epoch 6 is 0.1429131031036377 sec
Time for epoch 7 is 0.09333205223083496 sec
Time for epoch 8 is 0.1423022747039795 sec
Time for epoch 9 is 0.14398694038391113 sec
Time for epoch 10 is 0.14237475395202637 sec
Time for epoch 11 is 0.14293479919433594 sec
Time for epoch 12 is 0.09767651557922363 sec
Time for epoch 13 is 0.0964345932006836 sec
Time for epoch 14 is 0.14150214195251465 sec
Time for epoch 15 is 0.14276456832885742 sec
Time for epoch 16 is 0.09821438789367676 sec
Time for epoch 17 is 0.11067652702331543 sec
Time for epoch 18 is 0.09729409217834473 sec
Time for epoch 19 is 0.10210657119750977 sec
Time for epoch 20 is 0.14226508140563965 sec
Time for epoch 21 is 0.14197802543640137 sec
Time for epoch 22 is 0.1457366943359375 sec
Time for epoch 23 is 0.142

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 5.108618259429932 sec
Time for epoch 2 is 0.11729717254638672 sec
Time for epoch 3 is 0.14201831817626953 sec
Time for epoch 4 is 0.09353208541870117 sec
Time for epoch 5 is 0.14189815521240234 sec
Time for epoch 6 is 0.14312005043029785 sec
Time for epoch 7 is 0.14283418655395508 sec
Time for epoch 8 is 0.14236164093017578 sec
Time for epoch 9 is 0.10801458358764648 sec
Time for epoch 10 is 0.14216160774230957 sec
Time for epoch 11 is 0.14260292053222656 sec
Time for epoch 12 is 0.09064435958862305 sec
Time for epoch 13 is 0.14075303077697754 sec
Time for epoch 14 is 0.14270710945129395 sec
Time for epoch 15 is 0.09879946708679199 sec
Time for epoch 16 is 0.08747220039367676 sec
Time for epoch 17 is 0.09884834289550781 sec
Time for epoch 18 is 0.09754514694213867 sec
Time for epoch 19 is 0.14234185218811035 sec
Time for epoch 20 is 0.11502933502197266 sec
Time for epoch 21 is 0.1415879726409912 sec
Time for epoch 22 is 0.09510564804077148 sec
Time for epoch 23 is 0

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 5.108175754547119 sec
Time for epoch 2 is 0.17092537879943848 sec
Time for epoch 3 is 0.30669736862182617 sec
Time for epoch 4 is 0.1418008804321289 sec
Time for epoch 5 is 0.30440354347229004 sec
Time for epoch 6 is 0.3022449016571045 sec
Time for epoch 7 is 0.14568209648132324 sec
Time for epoch 8 is 0.3043818473815918 sec
Time for epoch 9 is 0.30393266677856445 sec
Time for epoch 10 is 0.1428220272064209 sec
Time for epoch 11 is 0.306225061416626 sec
Time for epoch 12 is 0.1429145336151123 sec
Time for epoch 13 is 0.30266451835632324 sec
Time for epoch 14 is 0.14223265647888184 sec
Time for epoch 15 is 0.14228296279907227 sec
Time for epoch 16 is 0.14341378211975098 sec
Time for epoch 17 is 0.09666037559509277 sec
Time for epoch 18 is 0.14271163940429688 sec
Time for epoch 19 is 0.1451585292816162 sec
Time for epoch 20 is 0.0985875129699707 sec
Time for epoch 21 is 0.14196252822875977 sec
Time for epoch 22 is 0.14213919639587402 sec
Time for epoch 23 is 0.1128129

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 5.108253717422485 sec
Time for epoch 2 is 0.10120224952697754 sec
Time for epoch 3 is 0.14210200309753418 sec
Time for epoch 4 is 0.14278197288513184 sec
Time for epoch 5 is 0.14321088790893555 sec
Time for epoch 6 is 0.0998995304107666 sec
Time for epoch 7 is 0.1417849063873291 sec
Time for epoch 8 is 0.14159297943115234 sec
Time for epoch 9 is 0.15357089042663574 sec
Time for epoch 10 is 0.14281964302062988 sec
Time for epoch 11 is 0.3067495822906494 sec
Time for epoch 12 is 0.30439066886901855 sec
Time for epoch 13 is 0.3026409149169922 sec
Time for epoch 14 is 0.14367318153381348 sec
Time for epoch 15 is 0.3048512935638428 sec
Time for epoch 16 is 0.1433863639831543 sec
Time for epoch 17 is 0.14325237274169922 sec
Time for epoch 18 is 0.30324602127075195 sec
Time for epoch 19 is 0.1427781581878662 sec
Time for epoch 20 is 0.30374741554260254 sec
Time for epoch 21 is 0.13555288314819336 sec
Time for epoch 22 is 0.3016355037689209 sec
Time for epoch 23 is 0.143480

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 2.7693228721618652 sec
Time for epoch 2 is 0.09161949157714844 sec
Time for epoch 3 is 0.09700155258178711 sec
Time for epoch 4 is 0.11616706848144531 sec
Time for epoch 5 is 0.1420445442199707 sec
Time for epoch 6 is 0.1416478157043457 sec
Time for epoch 7 is 0.14301276206970215 sec
Time for epoch 8 is 0.1476895809173584 sec
Time for epoch 9 is 0.14242815971374512 sec
Time for epoch 10 is 0.14228320121765137 sec
Time for epoch 11 is 0.09643840789794922 sec
Time for epoch 12 is 0.08832716941833496 sec
Time for epoch 13 is 0.14139986038208008 sec
Time for epoch 14 is 0.14388203620910645 sec
Time for epoch 15 is 0.14243841171264648 sec
Time for epoch 16 is 0.09074592590332031 sec
Time for epoch 17 is 0.14127445220947266 sec
Time for epoch 18 is 0.14237642288208008 sec
Time for epoch 19 is 0.10446023941040039 sec
Time for epoch 20 is 0.14266705513000488 sec
Time for epoch 21 is 0.1011960506439209 sec
Time for epoch 22 is 0.14235401153564453 sec
Time for epoch 23 is 0.1

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 3.365389347076416 sec
Time for epoch 2 is 0.3042945861816406 sec
Time for epoch 3 is 0.15984106063842773 sec
Time for epoch 4 is 0.30193376541137695 sec
Time for epoch 5 is 0.30254077911376953 sec
Time for epoch 6 is 0.30242466926574707 sec
Time for epoch 7 is 0.1399374008178711 sec
Time for epoch 8 is 0.14198613166809082 sec
Time for epoch 9 is 0.14256548881530762 sec
Time for epoch 10 is 0.14288830757141113 sec
Time for epoch 11 is 0.14219355583190918 sec
Time for epoch 12 is 0.14913082122802734 sec
Time for epoch 13 is 0.10282397270202637 sec
Time for epoch 14 is 0.09707474708557129 sec
Time for epoch 15 is 0.11193513870239258 sec
Time for epoch 16 is 0.1414506435394287 sec
Time for epoch 17 is 0.1439976692199707 sec
Time for epoch 18 is 0.14228034019470215 sec
Time for epoch 19 is 0.14254426956176758 sec
Time for epoch 20 is 0.14271926879882812 sec
Time for epoch 21 is 0.1452627182006836 sec
Time for epoch 22 is 0.09289956092834473 sec
Time for epoch 23 is 0.142

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 3.1037869453430176 sec
Time for epoch 2 is 0.14382672309875488 sec
Time for epoch 3 is 0.30339789390563965 sec
Time for epoch 4 is 0.14117813110351562 sec
Time for epoch 5 is 0.307065486907959 sec
Time for epoch 6 is 0.30316591262817383 sec
Time for epoch 7 is 0.15477228164672852 sec
Time for epoch 8 is 0.30650854110717773 sec
Time for epoch 9 is 0.3035905361175537 sec
Time for epoch 10 is 0.30548548698425293 sec
Time for epoch 11 is 0.3038957118988037 sec
Time for epoch 12 is 0.14220786094665527 sec
Time for epoch 13 is 0.12022662162780762 sec
Time for epoch 14 is 0.1430068016052246 sec
Time for epoch 15 is 0.14327168464660645 sec
Time for epoch 16 is 0.09375333786010742 sec
Time for epoch 17 is 0.1421070098876953 sec
Time for epoch 18 is 0.14138126373291016 sec
Time for epoch 19 is 0.10265517234802246 sec
Time for epoch 20 is 0.09000658988952637 sec
Time for epoch 21 is 0.14194011688232422 sec
Time for epoch 22 is 0.14356470108032227 sec
Time for epoch 23 is 0.142

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 5.1086742877960205 sec
Time for epoch 2 is 0.30318713188171387 sec
Time for epoch 3 is 0.30303454399108887 sec
Time for epoch 4 is 0.30519819259643555 sec
Time for epoch 5 is 0.3116273880004883 sec
Time for epoch 6 is 0.3050973415374756 sec
Time for epoch 7 is 0.3062725067138672 sec
Time for epoch 8 is 0.30257654190063477 sec
Time for epoch 9 is 0.3042185306549072 sec
Time for epoch 10 is 0.30440211296081543 sec
Time for epoch 11 is 0.3031914234161377 sec
Time for epoch 12 is 0.30314135551452637 sec
Time for epoch 13 is 0.14330244064331055 sec
Time for epoch 14 is 0.14584827423095703 sec
Time for epoch 15 is 0.11075639724731445 sec
Time for epoch 16 is 0.10483479499816895 sec
Time for epoch 17 is 0.10124635696411133 sec
Time for epoch 18 is 0.09473037719726562 sec
Time for epoch 19 is 0.10864973068237305 sec
Time for epoch 20 is 0.14146947860717773 sec
Time for epoch 21 is 0.10461759567260742 sec
Time for epoch 22 is 0.10163426399230957 sec
Time for epoch 23 is 0.11

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 5.1082305908203125 sec
Time for epoch 2 is 0.30391430854797363 sec
Time for epoch 3 is 0.30310511589050293 sec
Time for epoch 4 is 0.3039722442626953 sec
Time for epoch 5 is 0.14264583587646484 sec
Time for epoch 6 is 0.14183926582336426 sec
Time for epoch 7 is 0.10416245460510254 sec
Time for epoch 8 is 0.09455299377441406 sec
Time for epoch 9 is 0.1069650650024414 sec
Time for epoch 10 is 0.14125776290893555 sec
Time for epoch 11 is 0.14296340942382812 sec
Time for epoch 12 is 0.1428241729736328 sec
Time for epoch 13 is 0.10245871543884277 sec
Time for epoch 14 is 0.0946962833404541 sec
Time for epoch 15 is 0.09762406349182129 sec
Time for epoch 16 is 0.1022031307220459 sec
Time for epoch 17 is 0.09432363510131836 sec
Time for epoch 18 is 0.1410064697265625 sec
Time for epoch 19 is 0.14174199104309082 sec
Time for epoch 20 is 0.14850521087646484 sec
Time for epoch 21 is 0.14203524589538574 sec
Time for epoch 22 is 0.1424572467803955 sec
Time for epoch 23 is 0.1109

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 2.872126579284668 sec
Time for epoch 2 is 0.14233183860778809 sec
Time for epoch 3 is 0.10590958595275879 sec
Time for epoch 4 is 0.14141273498535156 sec
Time for epoch 5 is 0.09892606735229492 sec
Time for epoch 6 is 0.0901193618774414 sec
Time for epoch 7 is 0.09624338150024414 sec
Time for epoch 8 is 0.14258503913879395 sec
Time for epoch 9 is 0.14230942726135254 sec
Time for epoch 10 is 0.1427614688873291 sec
Time for epoch 11 is 0.14212465286254883 sec
Time for epoch 12 is 0.14319920539855957 sec
Time for epoch 13 is 0.14250779151916504 sec
Time for epoch 14 is 0.09619712829589844 sec
Time for epoch 15 is 0.14174389839172363 sec
Time for epoch 16 is 0.1411750316619873 sec
Time for epoch 17 is 0.0908346176147461 sec
Time for epoch 18 is 0.141279935836792 sec
Time for epoch 19 is 0.14269566535949707 sec
Time for epoch 20 is 0.1444401741027832 sec
Time for epoch 21 is 0.14188504219055176 sec
Time for epoch 22 is 0.1423969268798828 sec
Time for epoch 23 is 0.143638

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 5.10884165763855 sec
Time for epoch 2 is 0.16614031791687012 sec
Time for epoch 3 is 0.30125975608825684 sec
Time for epoch 4 is 0.09634947776794434 sec
Time for epoch 5 is 0.1425647735595703 sec
Time for epoch 6 is 0.09308910369873047 sec
Time for epoch 7 is 0.1420290470123291 sec
Time for epoch 8 is 0.10529875755310059 sec
Time for epoch 9 is 0.14166021347045898 sec
Time for epoch 10 is 0.09708261489868164 sec
Time for epoch 11 is 0.12236428260803223 sec
Time for epoch 12 is 0.09173941612243652 sec
Time for epoch 13 is 0.14208626747131348 sec
Time for epoch 14 is 0.09534859657287598 sec
Time for epoch 15 is 0.14150547981262207 sec
Time for epoch 16 is 0.11263227462768555 sec
Time for epoch 17 is 0.14101791381835938 sec
Time for epoch 18 is 0.11242341995239258 sec
Time for epoch 19 is 0.08546042442321777 sec
Time for epoch 20 is 0.14211702346801758 sec
Time for epoch 21 is 0.14403367042541504 sec
Time for epoch 22 is 0.1016230583190918 sec
Time for epoch 23 is 0.14

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 5.107357501983643 sec
Time for epoch 2 is 0.1434476375579834 sec
Time for epoch 3 is 0.14327764511108398 sec
Time for epoch 4 is 0.141845703125 sec
Time for epoch 5 is 0.14300918579101562 sec
Time for epoch 6 is 0.14996600151062012 sec
Time for epoch 7 is 0.3056364059448242 sec
Time for epoch 8 is 0.30306315422058105 sec
Time for epoch 9 is 0.3025786876678467 sec
Time for epoch 10 is 0.30236244201660156 sec
Time for epoch 11 is 0.306640625 sec
Time for epoch 12 is 0.1437077522277832 sec
Time for epoch 13 is 0.3041417598724365 sec
Time for epoch 14 is 0.1419532299041748 sec
Time for epoch 15 is 0.3047294616699219 sec
Time for epoch 16 is 0.14267683029174805 sec
Time for epoch 17 is 0.3036634922027588 sec
Time for epoch 18 is 0.13826966285705566 sec
Time for epoch 19 is 0.3055863380432129 sec
Time for epoch 20 is 0.14201807975769043 sec
Time for epoch 21 is 0.1429142951965332 sec
Time for epoch 22 is 0.14206409454345703 sec
Time for epoch 23 is 0.14261364936828613 sec

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 5.109131574630737 sec
Time for epoch 2 is 0.3033885955810547 sec
Time for epoch 3 is 0.30448126792907715 sec
Time for epoch 4 is 0.30257081985473633 sec
Time for epoch 5 is 0.31158447265625 sec
Time for epoch 6 is 0.3063631057739258 sec
Time for epoch 7 is 0.30299925804138184 sec
Time for epoch 8 is 0.15003013610839844 sec
Time for epoch 9 is 0.18790578842163086 sec
Time for epoch 10 is 0.14167571067810059 sec
Time for epoch 11 is 0.1116480827331543 sec
Time for epoch 12 is 0.09114193916320801 sec
Time for epoch 13 is 0.14224982261657715 sec
Time for epoch 14 is 0.10477232933044434 sec
Time for epoch 15 is 0.14182400703430176 sec
Time for epoch 16 is 0.1097865104675293 sec
Time for epoch 17 is 0.14221978187561035 sec
Time for epoch 18 is 0.14365673065185547 sec
Time for epoch 19 is 0.14813852310180664 sec
Time for epoch 20 is 0.14194512367248535 sec
Time for epoch 21 is 0.14649653434753418 sec
Time for epoch 22 is 0.10319852828979492 sec
Time for epoch 23 is 0.09322

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 2.925753593444824 sec
Time for epoch 2 is 0.1420280933380127 sec
Time for epoch 3 is 0.1421802043914795 sec
Time for epoch 4 is 0.14231586456298828 sec
Time for epoch 5 is 0.14133548736572266 sec
Time for epoch 6 is 0.09457921981811523 sec
Time for epoch 7 is 0.14242839813232422 sec
Time for epoch 8 is 0.14174699783325195 sec
Time for epoch 9 is 0.10983085632324219 sec
Time for epoch 10 is 0.10658407211303711 sec
Time for epoch 11 is 0.14159393310546875 sec
Time for epoch 12 is 0.0956563949584961 sec
Time for epoch 13 is 0.14269781112670898 sec
Time for epoch 14 is 0.09851312637329102 sec
Time for epoch 15 is 0.09103059768676758 sec
Time for epoch 16 is 0.141448974609375 sec
Time for epoch 17 is 0.1427440643310547 sec
Time for epoch 18 is 0.14301013946533203 sec
Time for epoch 19 is 0.09636998176574707 sec
Time for epoch 20 is 0.09493160247802734 sec
Time for epoch 21 is 0.14287924766540527 sec
Time for epoch 22 is 0.09960627555847168 sec
Time for epoch 23 is 0.1415

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 2.7909188270568848 sec
Time for epoch 2 is 0.09287357330322266 sec
Time for epoch 3 is 0.14245295524597168 sec
Time for epoch 4 is 0.14145851135253906 sec
Time for epoch 5 is 0.14246654510498047 sec
Time for epoch 6 is 0.1416482925415039 sec
Time for epoch 7 is 0.14342308044433594 sec
Time for epoch 8 is 0.14204788208007812 sec
Time for epoch 9 is 0.14150428771972656 sec
Time for epoch 10 is 0.14206933975219727 sec
Time for epoch 11 is 0.10194683074951172 sec
Time for epoch 12 is 0.08770918846130371 sec
Time for epoch 13 is 0.14240026473999023 sec
Time for epoch 14 is 0.14302635192871094 sec
Time for epoch 15 is 0.10891342163085938 sec
Time for epoch 16 is 0.1438596248626709 sec
Time for epoch 17 is 0.14254188537597656 sec
Time for epoch 18 is 0.14225316047668457 sec
Time for epoch 19 is 0.10520124435424805 sec
Time for epoch 20 is 0.0917806625366211 sec
Time for epoch 21 is 0.08421015739440918 sec
Time for epoch 22 is 0.1419079303741455 sec
Time for epoch 23 is 0.1

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 5.114576578140259 sec
Time for epoch 2 is 0.14174151420593262 sec
Time for epoch 3 is 0.10129761695861816 sec
Time for epoch 4 is 0.14164519309997559 sec
Time for epoch 5 is 0.14207124710083008 sec
Time for epoch 6 is 0.10392928123474121 sec
Time for epoch 7 is 0.08766651153564453 sec
Time for epoch 8 is 0.10276627540588379 sec
Time for epoch 9 is 0.14180731773376465 sec
Time for epoch 10 is 0.14795446395874023 sec
Time for epoch 11 is 0.1423795223236084 sec
Time for epoch 12 is 0.1449897289276123 sec
Time for epoch 13 is 0.09358453750610352 sec
Time for epoch 14 is 0.1420736312866211 sec
Time for epoch 15 is 0.09199237823486328 sec
Time for epoch 16 is 0.09638166427612305 sec
Time for epoch 17 is 0.1483321189880371 sec
Time for epoch 18 is 0.09598851203918457 sec
Time for epoch 19 is 0.10281968116760254 sec
Time for epoch 20 is 0.14212512969970703 sec
Time for epoch 21 is 0.1418933868408203 sec
Time for epoch 22 is 0.10564017295837402 sec
Time for epoch 23 is 0.105

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 5.109870910644531 sec
Time for epoch 2 is 0.14336872100830078 sec
Time for epoch 3 is 0.1148073673248291 sec
Time for epoch 4 is 0.08916401863098145 sec
Time for epoch 5 is 0.09970378875732422 sec
Time for epoch 6 is 0.1417245864868164 sec
Time for epoch 7 is 0.11010599136352539 sec
Time for epoch 8 is 0.1418132781982422 sec
Time for epoch 9 is 0.14254546165466309 sec
Time for epoch 10 is 0.14858770370483398 sec
Time for epoch 11 is 0.10175609588623047 sec
Time for epoch 12 is 0.14144492149353027 sec
Time for epoch 13 is 0.10427427291870117 sec
Time for epoch 14 is 0.11469531059265137 sec
Time for epoch 15 is 0.1053628921508789 sec
Time for epoch 16 is 0.14221978187561035 sec
Time for epoch 17 is 0.14173316955566406 sec
Time for epoch 18 is 0.14216113090515137 sec
Time for epoch 19 is 0.09250664710998535 sec
Time for epoch 20 is 0.14193487167358398 sec
Time for epoch 21 is 0.09096479415893555 sec
Time for epoch 22 is 0.14110374450683594 sec
Time for epoch 23 is 0.14

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 5.1090247631073 sec
Time for epoch 2 is 0.30303955078125 sec
Time for epoch 3 is 0.3045969009399414 sec
Time for epoch 4 is 0.14197444915771484 sec
Time for epoch 5 is 0.3074309825897217 sec
Time for epoch 6 is 0.3042762279510498 sec
Time for epoch 7 is 0.30420541763305664 sec
Time for epoch 8 is 0.3042159080505371 sec
Time for epoch 9 is 0.3029665946960449 sec
Time for epoch 10 is 0.1166234016418457 sec
Time for epoch 11 is 0.09382867813110352 sec
Time for epoch 12 is 0.10344362258911133 sec
Time for epoch 13 is 0.14197540283203125 sec
Time for epoch 14 is 0.09244728088378906 sec
Time for epoch 15 is 0.1417555809020996 sec
Time for epoch 16 is 0.0921475887298584 sec
Time for epoch 17 is 0.14241552352905273 sec
Time for epoch 18 is 0.14258980751037598 sec
Time for epoch 19 is 0.15007352828979492 sec
Time for epoch 20 is 0.1431443691253662 sec
Time for epoch 21 is 0.10119843482971191 sec
Time for epoch 22 is 0.0950784683227539 sec
Time for epoch 23 is 0.1422405242919

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 2.8986775875091553 sec
Time for epoch 2 is 0.1408402919769287 sec
Time for epoch 3 is 0.14474225044250488 sec
Time for epoch 4 is 0.09243631362915039 sec
Time for epoch 5 is 0.14178705215454102 sec
Time for epoch 6 is 0.14284825325012207 sec
Time for epoch 7 is 0.1117243766784668 sec
Time for epoch 8 is 0.09425234794616699 sec
Time for epoch 9 is 0.09378957748413086 sec
Time for epoch 10 is 0.3076634407043457 sec
Time for epoch 11 is 0.30240607261657715 sec
Time for epoch 12 is 0.305020809173584 sec
Time for epoch 13 is 0.30655694007873535 sec
Time for epoch 14 is 0.15147757530212402 sec
Time for epoch 15 is 0.3048560619354248 sec
Time for epoch 16 is 0.30260658264160156 sec
Time for epoch 17 is 0.303112268447876 sec
Time for epoch 18 is 0.14174342155456543 sec
Time for epoch 19 is 0.3038158416748047 sec
Time for epoch 20 is 0.15128445625305176 sec
Time for epoch 21 is 0.15404653549194336 sec
Time for epoch 22 is 0.3030364513397217 sec
Time for epoch 23 is 0.1046118

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 3.198951482772827 sec
Time for epoch 2 is 0.14169907569885254 sec
Time for epoch 3 is 0.10810017585754395 sec
Time for epoch 4 is 0.14135360717773438 sec
Time for epoch 5 is 0.12953424453735352 sec
Time for epoch 6 is 0.14307546615600586 sec
Time for epoch 7 is 0.1425166130065918 sec
Time for epoch 8 is 0.14225268363952637 sec
Time for epoch 9 is 0.14231657981872559 sec
Time for epoch 10 is 0.14301037788391113 sec
Time for epoch 11 is 0.17143535614013672 sec
Time for epoch 12 is 0.30371975898742676 sec
Time for epoch 13 is 0.30301547050476074 sec
Time for epoch 14 is 0.30239439010620117 sec
Time for epoch 15 is 0.15679121017456055 sec
Time for epoch 16 is 0.3018009662628174 sec
Time for epoch 17 is 0.30159497261047363 sec
Time for epoch 18 is 0.14427566528320312 sec
Time for epoch 19 is 0.30668067932128906 sec
Time for epoch 20 is 0.30353713035583496 sec
Time for epoch 21 is 0.30158424377441406 sec
Time for epoch 22 is 0.3031773567199707 sec
Time for epoch 23 is 0.1

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 3.099379777908325 sec
Time for epoch 2 is 0.30333709716796875 sec
Time for epoch 3 is 0.30927371978759766 sec
Time for epoch 4 is 0.1429152488708496 sec
Time for epoch 5 is 0.3031046390533447 sec
Time for epoch 6 is 0.143873929977417 sec
Time for epoch 7 is 0.3041863441467285 sec
Time for epoch 8 is 0.3031313419342041 sec
Time for epoch 9 is 0.30455732345581055 sec
Time for epoch 10 is 0.3044595718383789 sec
Time for epoch 11 is 0.15831375122070312 sec
Time for epoch 12 is 0.30742573738098145 sec
Time for epoch 13 is 0.1433553695678711 sec
Time for epoch 14 is 0.14704275131225586 sec
Time for epoch 15 is 0.1421680450439453 sec
Time for epoch 16 is 0.14638996124267578 sec
Time for epoch 17 is 0.1414954662322998 sec
Time for epoch 18 is 0.10565519332885742 sec
Time for epoch 19 is 0.14154696464538574 sec
Time for epoch 20 is 0.10176539421081543 sec
Time for epoch 21 is 0.14194393157958984 sec
Time for epoch 22 is 0.14400649070739746 sec
Time for epoch 23 is 0.14224910

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 5.1085121631622314 sec
Time for epoch 2 is 0.3032255172729492 sec
Time for epoch 3 is 0.09999322891235352 sec
Time for epoch 4 is 0.1417992115020752 sec
Time for epoch 5 is 0.10602712631225586 sec
Time for epoch 6 is 0.10611224174499512 sec
Time for epoch 7 is 0.12462663650512695 sec
Time for epoch 8 is 0.10718727111816406 sec
Time for epoch 9 is 0.1087942123413086 sec
Time for epoch 10 is 0.1161811351776123 sec
Time for epoch 11 is 0.14259648323059082 sec
Time for epoch 12 is 0.14232921600341797 sec
Time for epoch 13 is 0.14194798469543457 sec
Time for epoch 14 is 0.14220452308654785 sec
Time for epoch 15 is 0.12240481376647949 sec
Time for epoch 16 is 0.14166808128356934 sec
Time for epoch 17 is 0.1109309196472168 sec
Time for epoch 18 is 0.11179542541503906 sec
Time for epoch 19 is 0.1419684886932373 sec
Time for epoch 20 is 0.14325237274169922 sec
Time for epoch 21 is 0.1436765193939209 sec
Time for epoch 22 is 0.12438082695007324 sec
Time for epoch 23 is 0.1034

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 2.9399702548980713 sec
Time for epoch 2 is 0.1048429012298584 sec
Time for epoch 3 is 0.14151692390441895 sec
Time for epoch 4 is 0.09938430786132812 sec
Time for epoch 5 is 0.09466814994812012 sec
Time for epoch 6 is 0.14226460456848145 sec
Time for epoch 7 is 0.14354467391967773 sec
Time for epoch 8 is 0.10875391960144043 sec
Time for epoch 9 is 0.0997920036315918 sec
Time for epoch 10 is 0.09397530555725098 sec
Time for epoch 11 is 0.11838912963867188 sec
Time for epoch 12 is 0.1418306827545166 sec
Time for epoch 13 is 0.14216947555541992 sec
Time for epoch 14 is 0.3028578758239746 sec
Time for epoch 15 is 0.30420541763305664 sec
Time for epoch 16 is 0.15781140327453613 sec
Time for epoch 17 is 0.303250789642334 sec
Time for epoch 18 is 0.14190888404846191 sec
Time for epoch 19 is 0.3025379180908203 sec
Time for epoch 20 is 0.30629754066467285 sec
Time for epoch 21 is 0.3034062385559082 sec
Time for epoch 22 is 0.30638718605041504 sec
Time for epoch 23 is 0.30294

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 2.9494361877441406 sec
Time for epoch 2 is 0.14307832717895508 sec
Time for epoch 3 is 0.11776041984558105 sec
Time for epoch 4 is 0.11932921409606934 sec
Time for epoch 5 is 0.12342286109924316 sec
Time for epoch 6 is 0.09450221061706543 sec
Time for epoch 7 is 0.14080429077148438 sec
Time for epoch 8 is 0.1424403190612793 sec
Time for epoch 9 is 0.14258503913879395 sec
Time for epoch 10 is 0.09510517120361328 sec
Time for epoch 11 is 0.09269595146179199 sec
Time for epoch 12 is 0.12055706977844238 sec
Time for epoch 13 is 0.10294556617736816 sec
Time for epoch 14 is 0.10026359558105469 sec
Time for epoch 15 is 0.14529132843017578 sec
Time for epoch 16 is 0.10948348045349121 sec
Time for epoch 17 is 0.142012357711792 sec
Time for epoch 18 is 0.10833549499511719 sec
Time for epoch 19 is 0.08944129943847656 sec
Time for epoch 20 is 0.09874963760375977 sec
Time for epoch 21 is 0.10803627967834473 sec
Time for epoch 22 is 0.1421651840209961 sec
Time for epoch 23 is 0.1

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 5.111312389373779 sec
Time for epoch 2 is 0.12321209907531738 sec
Time for epoch 3 is 0.14191842079162598 sec
Time for epoch 4 is 0.14289259910583496 sec
Time for epoch 5 is 0.10563349723815918 sec
Time for epoch 6 is 0.1428055763244629 sec
Time for epoch 7 is 0.12018609046936035 sec
Time for epoch 8 is 0.10240030288696289 sec
Time for epoch 9 is 0.12006354331970215 sec
Time for epoch 10 is 0.11026310920715332 sec
Time for epoch 11 is 0.11844491958618164 sec
Time for epoch 12 is 0.14267849922180176 sec
Time for epoch 13 is 0.1061697006225586 sec
Time for epoch 14 is 0.09615540504455566 sec
Time for epoch 15 is 0.14122772216796875 sec
Time for epoch 16 is 0.14322948455810547 sec
Time for epoch 17 is 0.11872410774230957 sec
Time for epoch 18 is 0.14175200462341309 sec
Time for epoch 19 is 0.14270615577697754 sec
Time for epoch 20 is 0.1428835391998291 sec
Time for epoch 21 is 0.14222121238708496 sec
Time for epoch 22 is 0.09699606895446777 sec
Time for epoch 23 is 0.1

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 2.9320034980773926 sec
Time for epoch 2 is 0.14316534996032715 sec
Time for epoch 3 is 0.14627909660339355 sec
Time for epoch 4 is 0.10085773468017578 sec
Time for epoch 5 is 0.09666728973388672 sec
Time for epoch 6 is 0.14281940460205078 sec
Time for epoch 7 is 0.1418614387512207 sec
Time for epoch 8 is 0.14263916015625 sec
Time for epoch 9 is 0.1421365737915039 sec
Time for epoch 10 is 0.14672040939331055 sec
Time for epoch 11 is 0.14222288131713867 sec
Time for epoch 12 is 0.14770150184631348 sec
Time for epoch 13 is 0.30432796478271484 sec
Time for epoch 14 is 0.30479907989501953 sec
Time for epoch 15 is 0.30173230171203613 sec
Time for epoch 16 is 0.30411410331726074 sec
Time for epoch 17 is 0.3040792942047119 sec
Time for epoch 18 is 0.30608344078063965 sec
Time for epoch 19 is 0.30292654037475586 sec
Time for epoch 20 is 0.16507744789123535 sec
Time for epoch 21 is 0.303804874420166 sec
Time for epoch 22 is 0.3038642406463623 sec
Time for epoch 23 is 0.152116

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 5.10897421836853 sec
Time for epoch 2 is 0.30298542976379395 sec
Time for epoch 3 is 0.3023507595062256 sec
Time for epoch 4 is 0.304795503616333 sec
Time for epoch 5 is 0.3036196231842041 sec
Time for epoch 6 is 0.1425459384918213 sec
Time for epoch 7 is 0.146622896194458 sec
Time for epoch 8 is 0.14309453964233398 sec
Time for epoch 9 is 0.14659452438354492 sec
Time for epoch 10 is 0.09690070152282715 sec
Time for epoch 11 is 0.142486572265625 sec
Time for epoch 12 is 0.10635924339294434 sec
Time for epoch 13 is 0.14253473281860352 sec
Time for epoch 14 is 0.09324026107788086 sec
Time for epoch 15 is 0.14363789558410645 sec
Time for epoch 16 is 0.10088086128234863 sec
Time for epoch 17 is 0.14176702499389648 sec
Time for epoch 18 is 0.14278769493103027 sec
Time for epoch 19 is 0.14954495429992676 sec
Time for epoch 20 is 0.12910866737365723 sec
Time for epoch 21 is 0.11330747604370117 sec
Time for epoch 22 is 0.1413898468017578 sec
Time for epoch 23 is 0.117205619

/usr/local/lib/python3.8/dist-packages/keras/backend.py:5676: UserWarning: "`binary_crossentropy` received `from_logits=True`, but the `output` argument was produced by a Sigmoid activation and thus does not represent logits. Was this intended?
  output, from_logits = _get_logits(


Time for epoch 1 is 3.142324924468994 sec
Time for epoch 2 is 0.30512523651123047 sec
Time for epoch 3 is 0.14259839057922363 sec
Time for epoch 4 is 0.30179667472839355 sec
Time for epoch 5 is 0.1460108757019043 sec
Time for epoch 6 is 0.30315232276916504 sec
Time for epoch 7 is 0.30414509773254395 sec
Time for epoch 8 is 0.14205574989318848 sec
Time for epoch 9 is 0.3041188716888428 sec
Time for epoch 10 is 0.14101076126098633 sec
Time for epoch 11 is 0.301555871963501 sec
Time for epoch 12 is 0.30576348304748535 sec
Time for epoch 13 is 0.1423501968383789 sec
Time for epoch 14 is 0.14339447021484375 sec
Time for epoch 15 is 0.14197802543640137 sec
Time for epoch 16 is 0.14212989807128906 sec
Time for epoch 17 is 0.08864927291870117 sec
Time for epoch 18 is 0.14118742942810059 sec
Time for epoch 19 is 0.10097861289978027 sec
Time for epoch 20 is 0.14298486709594727 sec
Time for epoch 21 is 0.09817004203796387 sec
Time for epoch 22 is 0.1426987648010254 sec
Time for epoch 23 is 0.0930

In [ ]:
print('tempo: ',(tempo_gasto/60),' min' )
df = pd.DataFrame(testar, columns = ['amostras fake por classe'])
df['acc-media'] = array_media
df['acc-desv.pad.'] = array_std
df

tempo:  70.20661771694819  min


amostras fake por classe  acc-media  acc-desv.pad.
0                         4   0.816667       0.079495
1                         6   0.810417       0.076234

. zero dados fake, 24 dados reais por classe: (0.8216666666666669, 0.051934788169609614) rs = none

. 12 dados fake por classe, 12 dados reais por classe: (0.53125	0.098491) rs = none

. 24 dados fake por classe, 0 dados reais: ( )

.1000 dados fake por classe, 0 dados reais: ( )